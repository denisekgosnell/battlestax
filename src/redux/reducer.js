export const initialState = {
  initialized: false,
  gameId: 0,
  currentGame: {
    currentRound: 0,
    currentQuestion: 0,
    audienceSize: 0,
    players: [
      {
        name: "",
        vip: true,
        totalScore: 0,
      },
    ],
    rounds: [
      {
        id: 1,
        type: "",
        title: "Round One",
        scoreMultiplier: 1,
        questions: [
          {
            type: "",
            content: "",
            answers: [
              {
                name: "",
                answer: "",
                votes: [
                  {
                    name: "",
                  },
                ],
                audienceVotes: 0,
                score: 0,
              }
            ]
          }
        ]
      }
    ]
  }
};

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case 'INIT_GAME':
      // Changes the requesting field in state to true to show we are currently fetching to-dos
      return Object.assign({}, state,
        {
          initialized: true,
        },
      );
    default:
      return state;
  }
}