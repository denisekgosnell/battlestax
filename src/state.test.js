import reducer, { initialState } from "./state";

describe("start a new game reducer", () => {
  // TEST 0: testing the default returns the initialized state
  it("should return the initial state", () => {
    expect(reducer(initialState, {})).toMatchSnapshot();
  });

  // TEST 1: start a new game INIT_GAME
  it("should handle INIT_GAME request", () => {
    expect(
      reducer(initialState, {
        type: "INIT_GAME",
      })
    ).toMatchSnapshot();
  });
  // TEST 2: should add 1 player, and that player should be the vip
  it("should handle ADD_PLAYER request", () => {
    expect(
      reducer(initialState, {
        type: "ADD_PLAYER",
      })
    ).toMatchSnapshot();
  });
  // TEST 4: should add 3 players
  // TEST 5: should start a game with at least 3 players
  // TEST 6: should start a game with no more than 8 players
  // TEST 7: should allow audience members
  // TEST 8: should allow the game to start
  // TEST 9: should move to the game tutorial
  it("should handle ADD_PLAYER request", () => {
    expect(
      reducer(initialState, {
        type: "START_TUTORIAL",
      })
    ).toMatchSnapshot();
  });
  // TEST 10: should move to a question round
  // TEST 11: should receive question responses from players
  // TEST 12: should move to a voting round
  // TEST 13: should receive votes from players
  // TEST 14: should receive votes from the audience
  // TEST 15: should update the leaderboard
  // TEST 16: should move to the leaderboard
  // TEST 17: should move to the end game screen
  // TEST 18: should move to the start game screen
});
