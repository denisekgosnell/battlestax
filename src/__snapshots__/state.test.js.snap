// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`start a new game reducer should handle ADD_PLAYER request 1`] = `
Object {
  "currentGame": Object {
    "audienceSize": 0,
    "currentQuestion": 0,
    "currentRound": 0,
    "players": Array [
      Object {
        "name": undefined,
        "totalScore": 0,
        "vip": true,
      },
    ],
    "rounds": Array [
      Object {
        "id": 1,
        "questions": Array [
          Object {
            "answers": Array [
              Object {
                "answer": "",
                "audienceVotes": 0,
                "name": "",
                "score": 0,
                "votes": Array [
                  Object {
                    "name": "",
                  },
                ],
              },
            ],
            "content": "",
            "type": "",
          },
        ],
        "scoreMultiplier": 1,
        "title": "Round One",
        "type": "",
      },
    ],
  },
  "gameId": 0,
  "initialized": false,
}
`;

exports[`start a new game reducer should handle ADD_PLAYER request 2`] = `
Object {
  "currentGame": Object {
    "audienceSize": 0,
    "currentQuestion": 0,
    "currentRound": "TUTORIAL_ROUND",
    "players": Array [],
    "rounds": Array [
      Object {
        "id": 1,
        "questions": Array [
          Object {
            "answers": Array [
              Object {
                "answer": "",
                "audienceVotes": 0,
                "name": "",
                "score": 0,
                "votes": Array [
                  Object {
                    "name": "",
                  },
                ],
              },
            ],
            "content": "",
            "type": "",
          },
        ],
        "scoreMultiplier": 1,
        "title": "Round One",
        "type": "",
      },
    ],
  },
  "gameId": 0,
  "initialized": false,
}
`;

exports[`start a new game reducer should handle INIT_GAME request 1`] = `
Object {
  "currentGame": Object {
    "audienceSize": 0,
    "currentQuestion": 0,
    "currentRound": 0,
    "players": Array [],
    "rounds": Array [
      Object {
        "id": 1,
        "questions": Array [
          Object {
            "answers": Array [
              Object {
                "answer": "",
                "audienceVotes": 0,
                "name": "",
                "score": 0,
                "votes": Array [
                  Object {
                    "name": "",
                  },
                ],
              },
            ],
            "content": "",
            "type": "",
          },
        ],
        "scoreMultiplier": 1,
        "title": "Round One",
        "type": "",
      },
    ],
  },
  "gameId": 0,
  "initialized": true,
}
`;

exports[`start a new game reducer should return the initial state 1`] = `
Object {
  "currentGame": Object {
    "audienceSize": 0,
    "currentQuestion": 0,
    "currentRound": 0,
    "players": Array [],
    "rounds": Array [
      Object {
        "id": 1,
        "questions": Array [
          Object {
            "answers": Array [
              Object {
                "answer": "",
                "audienceVotes": 0,
                "name": "",
                "score": 0,
                "votes": Array [
                  Object {
                    "name": "",
                  },
                ],
              },
            ],
            "content": "",
            "type": "",
          },
        ],
        "scoreMultiplier": 1,
        "title": "Round One",
        "type": "",
      },
    ],
  },
  "gameId": 0,
  "initialized": false,
}
`;
